python-modules.sh

```diff
@@ -7,21 +7,35 @@ requires:
 build_requires:
   - curl
 env:
-  SSL_CERT_FILE: "$(export PYTHONPATH=$PYTHON_MODULES_ROOT/lib/python2.7/site-packages:$PYTHONPATH; export PATH=$PYTHON_ROOT/bin:$PATH; export LD_LIBRARY_PATH=$PYTHON_ROOT/lib:$LD_LIBRARY_PATH; python -c \"import certifi; print certifi.where()\")"
+  SSL_CERT_FILE: "$(env PYTHONPATH=$PYTHON_MODULES_ROOT/lib/python$(python -c \"import distutils.sysconfig; print(distutils.sysconfig.get_python_version())\")/site-packages:$PYTHONPATH python -c \"import certifi; print certifi.where()\")"
 prepend_path:
   PYTHONPATH: $PYTHON_MODULES_ROOT/lib/python2.7/site-packages:$PYTHONPATH
 prefer_system: (?!slc5)
 prefer_system_check:
-  python -c 'import matplotlib; import numpy; import certifi'
+  python -c 'import matplotlib; import numpy; import certifi; import IPython; import ipykernel; import yaml'
 ---
 #!/bin/bash -ex
 
-# Install extra packages with pip
-pip install --install-option="--prefix=$INSTALLROOT" "mock==1.0.0"
-pip install --install-option="--prefix=$INSTALLROOT" "numpy==1.9.2"
-pip install --install-option="--prefix=$INSTALLROOT" "certifi==2015.9.6.2"
+# Force pip installation of packages found in current PYTHONPATH
+unset PYTHONPATH
 
-# Install matplotlib (very tricky)
+# The X.Y in pythonX.Y
+export PYVER=$(python -c 'import distutils.sysconfig; print(distutils.sysconfig.get_python_version())')
+
+# Install as much as possible with pip
+export PYTHONUSERBASE=$INSTALLROOT
+for X in "mock==1.0.0"         \
```

Comment:
Unfortunately `pip` does not fail (exitcode != 0) if at least one of the packages installed correctly. So we have to separate the `pip`s in order to catch potential errors.